src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.OSX.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace System.Net.Sockets
+{
+    internal static partial class SocketPal
+    {
+        public const bool SupportsMultipleConnectAttempts = false;
+
+        public static void PrimeForNextConnectAttempt(int fileDescriptor, int socketAddressLen)
```

Comment:
One interesting note w.r.t. this change: C# does not allow access modifiers on partial methods and also requires that all definitions agree on `unsafe`. Therefore, the primary definition (i.e. the definition in SocketPal.Unix.cs) is written `static unsafe partial void ...`